pkg/context/context.go

```diff
@@ -116,7 +116,7 @@ func (ctx *Context) Teardown() error {
 		// then it is also in charge of the cleanup.
 		return nil
 	}
-	return snapshot.Cleanup(ctx.SnapshotRoot)
+	return snapshot.Cleanup(ctx.Chroot)
```

Comment:
In general I don't think we should clean up `Chroot`. While this should work in the happy path, we leave too much room for error: for example we need to ensure that we never try to clean up user-supplied directory with user data.
Now: the original code was ALSO wrong, and for extra fun the doc is also a bit misleading. I'll fix the latter with a separate PR.

Reply:
https://github.com/jaypipes/ghw/pull/239

Reply:
agreed, looks like `Chroot` choice may be too risky. I was going to add another flag to ctx (cleanupNeeded) and use it for cleanup, but it didn't look elegant enough.

Reply:
I'd store the internally-managed directory explicitely, like in...
```go
type Context struct {
	Chroot            string
	EnableTools       bool
	SnapshotPath      string
	SnapshotRoot      string // set by user
	SnapshotExclusive bool
	alert             option.Alerter
	parent            *Context
	snapshotRoot      string // managed internally
}
//...
func (ctx *Context) Setup() error {
	if ctx.SnapshotPath == "" {
		// nothing to do!
		return nil
	}

	var err error
	root := ctx.SnapshotRoot
	if root == "" {
		root, err = snapshot.Unpack(ctx.SnapshotPath)
		ctx.snapshotRoot = root
	} else {
		var flags uint
		if ctx.SnapshotExclusive {
			flags |= snapshot.OwnTargetDirectory
		}
		_, err = snapshot.UnpackInto(ctx.SnapshotPath, root, flags)
	}
	if err != nil {
		return err
	}

	ctx.Chroot = root
	return nil
}

func (ctx *Context) Teardown() error {
	if ctx.snapshotRoot == "" {
		// if the client code provided the unpack directory,
		// then it is also in charge of the cleanup.
		return nil
	}
	return snapshot.Cleanup(ctx.snapshotRoot)
}
```

Reply:
yes,  maybe indeed it is worth adding a new internal field just to be safe. I'll post the changes soon, thanks.

Reply:
Thanks! no hurry. And with my https://github.com/jaypipes/ghw/pull/239 merged, is clear that ghw wants to clean only the directory it owns - hence when `GHW_SNAPSHOT_ROOT` is *not* given. This should help a bit with this PR as well.

Reply:
I re-pushed following your comments, and yes, now the code is cleaner, thanks for the review.